tests/functional/relation_quoting/test_relation_quoting.py

```diff
@@ -0,0 +1,64 @@
+import pytest
+
+from dbt.tests.util import read_file, run_dbt
+
+_SOURCES_YML = """
+sources:
+  - name: source_name
+    database: source_database
+    schema: source_schema
+    tables:
+      - name: customers
+"""
+
+
+class TestSourceQuotingGlobalConfigs:
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        # Postgres quoting configs are True by default -- turn them all to False to show they are not respected during source rendering
+        return {
+            "quoting": {
+                "database": False,
+                "schema": False,
+                "identifier": False,
+            },
+        }
+
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "sources.yml": _SOURCES_YML,
+            "model.sql": "select * from {{ source('source_name', 'customers') }}",
+        }
+
+    def test_sources_ignore_global_quoting_configs(self, project):
+        run_dbt(["compile"])
+
+        generated_sql = read_file("target", "compiled", "test", "models", "model.sql")
+        assert generated_sql == 'select * from "source_database"."source_schema"."customers"'
+
```

Comment:
confirmed the above test does fail on main as expected: 

```sh
tests/functional/relation_quoting/test_relation_quoting.py F.                                                                                         [100%]

========================================================================= FAILURES ==========================================================================
_________________________________________ TestSourceQuotingGlobalConfigs.test_sources_ignore_global_quoting_configs _________________________________________

self = <test_relation_quoting.TestSourceQuotingGlobalConfigs object at 0x114bdba30>
project = <dbt.tests.fixtures.project.TestProjInfo object at 0x115b1e9b0>

    def test_sources_ignore_global_quoting_configs(self, project):
        run_dbt(["compile"])
    
        generated_sql = read_file("target", "compiled", "test", "models", "model.sql")
>       assert generated_sql == 'select * from "source_database"."source_schema"."customers"'
E       assert 'select * fro...ema.customers' == 'select * fro..."."customers"'
E         - select * from "source_database"."source_schema"."customers"
E         ?               -               - -             - -         -
E         + select * from source_database.source_schema.customers

/Users/michelleark/src/dbt-core/tests/functional/relation_quoting/test_relation_quoting.py:38: AssertionError
------------------------------------------------------------------- Captured stdout setup -------------------------
```

Reply:
Have also confirmed these tests pass with no code changes on 1.7.latest: https://github.com/dbt-labs/dbt-core/pull/10906